VolleyPaint/Assets/Scripts/ProjectileLaunch.cs

```diff
@@ -0,0 +1,136 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Blobcreate.Universal;
+using Blobcreate.ProjectileToolkit;
+
+[CreateAssetMenu(menuName = "Litkey/Ability/Projectile/ManageBall")]
+public class ProjectileLaunch : Ability
+{
+	[Header("Specific Settings")]
+	//public Transform launchPoint;
+	//public Rigidbody bulletPrefab;
+	public LayerMask groundMask;
+	public float torqueForce = 0f;
+	public float smallA = -0.1f;
+	public float bigA = -0.01f;
+	public float lerpSpeed = 5f;
+	private TrajectoryPredictor trajectory;
+	public bool drawLine;
+	public bool faceDirection;
+	float currentA;
+	float currentTorque;
+
+	private GameObject ball;
+	private Rigidbody rb;
+
+	//private TrajectoryPredictor trajectorySpawned;
+
+	// Player shoots some kind of projectile or ray
+	// if it hits the ball, stop the ball for a while, player gains access to shoot the ball in the direction player is looking at
+	// and shoot the ball
+    public override void UseAbility(GameObject parent)
+    {
+        base.UseAbility(parent);
+		// Shoot projectile to the ball
+		//var b = Instantiate(bulletPrefab, launchPoint.position, launchPoint.rotation);
+		//Physics.Raycast(Camera.main.ScreenPointToRay(Input.mousePosition), out var hit, 300f, groundMask);
+		rb.velocity = Vector3.zero;
+		rb.isKinematic = true;
+	}
+
+    public override void OnAbilityStart(GameObject parent)
+    {
+        base.OnAbilityStart(parent);
+		ball = GameObject.FindGameObjectWithTag("Ball");
+
+		if (ball != null) 
+		{
+			rb = ball.GetComponent<Rigidbody>();
+			ball.GetComponent<ProjectileBehaviour>().faceDirection = faceDirection;
+		}
+		currentA = smallA;
+		OnFireButtonDown();
+
+		if (drawLine)
+        {
+			if (ball != null)
+            {
+				trajectory = ball.GetComponentInChildren<TrajectoryPredictor>();
+				trajectory.enabled = false;
+            }
+        }
+	}
+
+    public override void OnAbilityRunning(GameObject parent)
+    {
+        base.OnAbilityRunning(parent);
+		OnFireButton();
+		DrawPredictionLine();
+	}
+
+	// fire the projectile
+    public override void OnAbilityEnd(GameObject parent)
+    {
+        base.OnAbilityEnd(parent);
+		OnFireButtonUp();
+		trajectory.enabled = false;
+	}
+
+	private void DrawPredictionLine()
+    {
+		if (drawLine)
+        {
+			Physics.Raycast(Camera.main.ScreenPointToRay(Input.mousePosition),
+						out var hit, 300f, groundMask);
+			RenderLaunch(ball.transform.position, hit.point);
+			trajectory.enabled = true;
+        }
+	}
+	public void RenderLaunch(Vector3 origin, Vector3 target)
```

Comment:
if you could add comments to this when u get the chance, that would be really helpful for me to understand it a bit better :)

Reply:
Sure. I'm sorry I missed it.